frontend/src/Pages/Customer/BuyerCatalog.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import axios from 'axios';
+import { Row, Col, Button, Container, Jumbotron, Card, Image, ListGroup } from 'react-bootstrap';
+import ROUTES from '../../routes';
+
+const CALL_BUTTON_IMAGE_URL = "https://secure.webtoolhub.com/static/resources/icons/set112/56bb7adb.png";
+
+class BuyerCatalog extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      shopID: props.match.params.shopid,
+      shopFetched: false
+    };
+  }
+
+  componentDidMount() {
+    this.getShopDetails();
+  }
+
+  getShopDetails() {
+    const shopID = this.state.shopID;
+    axios.get(ROUTES.api.get.shopByShopID.replace("%b", shopID))
+      .then((response) => {
+        response.data["shopFetched"] = true;
```

Comment:
this is not clear, what are we setting in state, can we be more explicit and naming variables here since its not obvious

Reply:
Fixed this. Thanks a lot!!!